src/partition/ParquetInterface.cc

```diff
@@ -0,0 +1,353 @@
+
+    arrow::Status status = Table2CSVBuffer(table_loc, buffSize, buf);
+
+    if (status.ok()) return arrow::Status::OK();
+
+    return arrow::Status::ExecutionError("Error while writing table to CSV buffer");
+}
+
+// Create char buffer containing the table in csv format
+arrow::Status ParquetFile::Table2CSVBuffer(std::shared_ptr<arrow::Table>& table, int& buffSize, void* buf) {
+    ARROW_ASSIGN_OR_RAISE(auto outstream, arrow::io::BufferOutputStream::Create(1 << 10));
+
+    // Options : null string, no header, no quotes around strings
+    arrow::csv::WriteOptions writeOpt = arrow::csv::WriteOptions::Defaults();
+    writeOpt.null_string = "\\N";
+    writeOpt.include_header = false;
+    writeOpt.quoting_style = arrow::csv::QuotingStyle::None;
+
+    ARROW_RETURN_NOT_OK(arrow::csv::WriteCSV(*table, writeOpt, outstream.get()));
+    ARROW_ASSIGN_OR_RAISE(auto buffer, outstream->Finish());
+
+    // auto buffer_ptr = buffer.get()->data();
+    buffSize = buffer->size();
+    std::cout << "ParquetFile::Table2CSVBuffer - buffer length : " << buffSize << " // " << m_maxBufferSize
+              << std::endl;
+
+    memcpy(buf, (void*)buffer.get()->data(), buffer->size());
+
+    //  auto myfile = std::fstream("buffer_"+std::to_string(m_batchNumber)+".csv", std::ios::out);
+    //  myfile.write((char*)&buffer_ptr[0], buffSize);
+    //  myfile.close();
+
+    return arrow::Status::OK();
+}
+
+// Read a batch from the file and format the table iaccording to the partition configuration file
+//    -> column reordering, true/false -> 1/0, remove null values, ...
+arrow::Status ParquetFile::ReadNextBatchTable_Formatted(std::shared_ptr<arrow::Table>& outputTable) {
+    arrow::Result<std::shared_ptr<arrow::RecordBatch>> maybe_batch = m_rb_reader_gbl->Next();
+
+    std::vector<std::string> paramNotFound;
+    std::map<std::string, std::shared_ptr<arrow::Field>> fieldConfig;
+
+    if (maybe_batch != nullptr) {
+        // DumpProcessMemory("VmRSS", true);
+
+        ARROW_ASSIGN_OR_RAISE(auto batch, maybe_batch);
+        std::shared_ptr<arrow::Table> initTable;
+        ARROW_ASSIGN_OR_RAISE(initTable, arrow::Table::FromRecordBatches(batch->schema(), {batch}));
+
+        // Increment the batch number
+        m_batchNumber++;
+
+        const arrow::FieldVector fields = initTable->schema()->fields();
+        for (auto fd : fields) {
+            fieldConfig[fd->name()] = fd;
+        }
+
+        arrow::FieldVector formatedTable_fields;
+        std::vector<std::shared_ptr<arrow::ChunkedArray>> formatedTable_columns;
+
+        // Loop over the column names as defined in the partition config file
+        for (std::string paramName : m_parameterNames) {
+            std::shared_ptr<arrow::ChunkedArray> chunkedArray = initTable->GetColumnByName(paramName);
+
+            // Column not found in the arrow table...
+            if (chunkedArray == NULLPTR) {
```

Comment:
What is `NULLPTR`? Is this some type specific to Arrow API? Otherwise, it needs to be low-case `nullptr`.